### DISCUSSION TITLE
Extracting data from Powerpoint-generated slide-show PDF

### DISCUSSION THREAD

**Discussion by petermr** [jsvine/pdfplumber#1186] [Ask for help with specific PDFs]
I have been sent a set of "PDF slides" presumably generated from Powerpoint. I want to extract as much as possible, especially text but also images.

Looks like it's one page per slide so looks like I can use PDFPlumber naturally. 

I can make an example available if possible.

**Comment by jsvine** [comment]
As far as I understand it, a PDF generated by slideshow software shouldn't be much different (at least in an abstract sense) from a PDF generated by other software. But perhaps you're seeing some challenging differences?

**Reply by petermr** [reply]
Thanks!
Yes, you're right. It's simpler than I thought - each slide is a PDF page. Since we can process these with `pdfplumber` it's straightforward to loop over them. (I don't know how the PDF manages the sidebar of thumbnails - maybe it's all done by the viewer.)

Interestingly, and counter-intuitively, If I want to extract information from PPTX documents it's easier in the current case to dump them as PDF and then use `pdfplumber` . PPTX has more semantics but it creates zillions of files and relationships. But it does preserve images as JPEGs and I am not sure how well pdfplumber can extract images from the PDF stream.
